### PR TITLE
Wi-Fi shell improvs

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -1655,7 +1655,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 	SHELL_CMD_ARG(disconnect, NULL, "Disconnect from the Wi-Fi AP",
 		  cmd_wifi_disconnect,
 		  1, 0),
-	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F PS on/off, no arguments will dump config",
+	SHELL_CMD_ARG(ps, NULL, "Configure Wi-F power save state\n"
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c]: Configure or display Wi-Fi power save state\n

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Comment by krish2718** [comment]
FYI, for a single line change in help, had to pull in many of the scan re-work fromtree commits, as upmerge is still pending.

**Review by sachinthegreen — APPROVED** [review]

File: include/zephyr/net/wifi_mgmt.h [hunk 2]
```diff
@@ -41,6 +41,12 @@ extern "C" {
 #define WIFI_MGMT_SCAN_SSID_FILT_MAX 0
 #endif /* CONFIG_WIFI_MGMT_SCAN_SSID_FILT_MAX */
 
+#ifdef CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+#define WIFI_MGMT_SCAN_CHAN_MAX_MANUAL CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+#else
+#define WIFI_MGMT_SCAN_CHAN_MAX_MANUAL 1
+#endif /* CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL */
```
Comment (trantanen) [inline include/zephyr/net/wifi_mgmt.h:48]: In which case we hit the else branch? Can there be some other `wifi_mgmt` implementation than the one in `subsys/net/l2/wifi`?

Comment (krish2718) [reply include/zephyr/net/wifi_mgmt.h:48]: This is a workaround for a bug in few cellular/location samples, the Wi-Fi headers are included unconditionally (`nrf/include/net/wifi_location_common.h`) and if the Wi-Fi isn't enabled, this poses a problem, as Kconfig isn't included. This was first instance of a Kconfig being used in the header files.

Comment (trantanen) [reply include/zephyr/net/wifi_mgmt.h:48]: Thanks for the info. I know some/many people prefer to use as little conditions as possible which causes these issues. I'll keep this in mind to see if there would be better solution for this.

**Review by rado17 — APPROVED** [review]

**Pull request merged**